src/services/interceptors/ServerStatusInterceptor.ts

```diff
@@ -0,0 +1,102 @@
+import {App} from "../../ManagementConsole";
+
+const module: ng.IModule = App.module("managementConsole.services.interceptors", [
+  "managementConsole.services.authentication"
+]);
+
+export class ServerStatusInterceptor {
+
+  static $inject: string[] = ["$injector", "$q"];
+
+  public static self: any = ServerStatusInterceptor;
+
+  public static serverStatusTimeout:number = 5000;
+  public static serverStatusTimer: any;
+  public static statusCallsFired: number = 0;
+  public static maxStatusCalls: number = 20;
+
+  public static ngInjector:any;
+  public static ngQ:any;
+
+  public static resetLastRequestTimer(timer: number, self: any): any {
+    if (timer) {
+      clearTimeout(timer);
+    }
+
+    return setTimeout(function(): void {
+      self.ngInjector.get("availabilityCheck").checkServerIsAlive();
```

Comment:
As you have now exposed AvailabilityCheck as a service, why can't you just inject the service via $inject?